ehsm_kms_service/secret_manager_apis.js

```diff
@@ -0,0 +1,223 @@
+}
+
+const queryDefaultCMK = (appid) => {
+    return {
+        selector: {
+            appid
+        },
+        fields: ['sm_default_cmk'],
+        limit: 1
+    }
+}
+
+//create Secret
+const createSecret = async (res, appid, payload, DB) => {
+    try {
+        let secretData = getParam_String(payload, 'secretData')
+        let secretName = getParam_String(payload, 'secretName')
+        let encryptionKeyId = getParam_String(payload, 'encryptionKeyId', false)
+        let description = getParam_String(payload, 'description')
+        let rotationInterval = getParam_String(payload, 'rotationInterval', false)
+        const createTime = new Date().getTime()
+        let keyBlob = ''
+        let nextRotationDate = ''
+        if (!checkStringParam(secretData, true)) {
+            res.send(_result(400, 'secretData cannot be empty and must be string'))
+            return
+        }
+        if (!checkStringParam(secretName, true)) {
+            res.send(_result(400, 'secretName cannot be empty and must be string'))
+            return
+        }
+        if (!checkStringParam(description, false)) {
+            res.send(_result(400, 'description must be string'))
+            return
+        }
+        if (!checkStringParam(rotationInterval, false)) {
+            res.send(_result(400, 'rotationInterval must be string'))
+            return
+        }
+        if (!check_encryptionKeyId_format(encryptionKeyId)) {
+            res.send(_result(400, 'encryptionKeyId format wrong'))
+            return
+        }
+        if (!check_rotationInterval_format(rotationInterval)) {
+            res.send(_result(400, 'rotationInterval format wrong'))
+            return
+        }
+
+        const secret_name_query = {
+            selector: {
+                appid,
+                secretName
+            },
+            fields: ['appid', 'secretName'],
+            limit: 1,
+        }
+        let secret_name_result = await DB.partitionedFind('secret_metadata', secret_name_query)
+        if (secret_name_result.docs.length > 0) {
+            res.send(_result(400, 'Secret name already exists, cannot use the same secret name'))
+            return
+        }
+
+        if (encryptionKeyId != '' && typeof (encryptionKeyId) == 'string') {
+            let key_id_res = await DB.partitionedFind('cmk', queryKeyId(appid, encryptionKeyId))
+            if (key_id_res.docs.length == 0) {
+                res.send(_result(400, 'Cannot find key id'))
+                return
+            } else {
+                keyBlob = key_id_res.docs[0].keyBlob
+            }
+        } else if (encryptionKeyId == "" || encryptionKeyId == undefined) {
+            let default_cmk_res = await DB.partitionedFind('user_info', queryDefaultCMK(appid))
+            if (default_cmk_res.docs.length == 0) {
+                res.send(_result(400, 'Cannot find default CMK'))
+                return
+            }
+            keyBlob = default_cmk_res.docs[0].sm_default_cmk
```

Comment:
so the sm_default_cmk is a keyblob rather than a cmk_id?

Reply:
sm_ default_ CMK is used for encryption as keyblob here